client/src/templates/Challenges/components/help-modal.tsx

```diff
@@ -53,61 +89,210 @@ function HelpModal({
   challengeBlock,
   challengeTitle
 }: HelpModalProps): JSX.Element {
+  const [showHelpForm, setShowHelpForm] = useState(false);
+  const [description, setDescription] = useState('');
+  const [readSearchCheckbox, setReadSearchCheckbox] = useState(false);
+  const [similarQuestionsCheckbox, setSimilarQuestionsCheckbox] =
+    useState(false);
+  const DESCRIPTION_MIN = 50;
+  const DESCRIPTION_MAX = 500;
+  const formRef = useRef<HTMLFormElement>(null);
+
+  useEffect(() => {
+    if (showHelpForm) {
+      formRef.current?.querySelector('input')?.focus();
+    }
+  }, [showHelpForm]);
+
+  const canSubmitForm = useCallback(() => {
+    return (
+      description.length >= DESCRIPTION_MIN &&
+      readSearchCheckbox &&
+      similarQuestionsCheckbox
+    );
+  }, [description, readSearchCheckbox, similarQuestionsCheckbox]);
+
+  const resetFormValues = () => {
+    setDescription('');
+    setReadSearchCheckbox(false);
+    setSimilarQuestionsCheckbox(false);
+  };
+
+  const handleCheckboxChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    set: React.Dispatch<React.SetStateAction<boolean>>
+  ) => {
+    set(event.target.checked);
+  };
+
   if (isOpen) {
     callGA({ event: 'pageview', pagePath: '/help-modal' });
   }
   return (
-    <Modal dialogClassName='help-modal' onHide={closeHelpModal} show={isOpen}>
+    <Modal
+      dialogClassName='help-modal'
+      onHide={() => {
+        closeHelpModal();
+        setShowHelpForm(false);
+        resetFormValues();
+      }}
+      show={isOpen}
+    >
       <Modal.Header className='help-modal-header fcc-modal' closeButton={true}>
         <Modal.Title className='text-center'>
```

Comment:
```suggestion
      aria-labelledby='ask-for-help-modal'
    >
      <Modal.Header className='help-modal-header fcc-modal' closeButton={true}>
        <Modal.Title id='ask-for-help-modal' className='text-center'>
```

I think we should give this dialog a programmatic (accessible) name. Currently, JAWS is just announcing "dialog". If we give the dialog a name, JAWS will announce "Ask for help dialog". I think this helps clarify that a dialog was opened specifically for the purpose of asking for help.